mam/v2/mam.c

```diff
@@ -300,16 +307,22 @@ retcode_t mam2_send_msg(mam2_send_msg_context *cfg, trits_t *msg) {
       err_guard(!(trits_size(*msg) < msg_size), RC_MAM2_BUFFER_TOO_SMALL);
     b = msg;
 
-    if (cfg->ep)
-      mss_skn(cfg->ep->m, skn);
-    else
-      mss_skn(cfg->ch->m, skn);
+    // FIXME (@tsvisabo) - uncomment
+    if (cfg->ep) {
+      // mss_skn(cfg->ep->m, skn);
+    } else {
+      // mss_skn(cfg->ch->m, skn);
+    }
 
     // TODO Remove when mam handles flex_trits
-    trits_t cfg_ch = mam2_channel_name(cfg->ch);
-    TRIT_ARRAY_MAKE_FROM_RAW(ch, cfg_ch.n - cfg_ch.d, cfg_ch.p + cfg_ch.d);
-    trits_t cfg_ep = cfg->ep ? mam2_endpoint_name(cfg->ep) : trits_null();
-    TRIT_ARRAY_MAKE_FROM_RAW(ep, cfg_ep.n - cfg_ep.d, cfg_ep.p + cfg_ep.d);
+    trit_array_t ch = mam2_channel_name(cfg->ch);
+    trit_array_t ep;
+    if (cfg->ep) {
+      ep = mam2_endpoint_name(cfg->ep);
+    } else {
+      ep.num_trits = 0;
+      ep.trits = 0;
```

Comment:
Maybe set `ep.num_bytes = 0` as well in case we rely on it ?
Also, `ep.trits = NULL;`